### PR TITLE
fix(BA-6994): split reserved start time from execution start time on sessions

### PR DESCRIPTION
**Pull request by HyeockJinKim**
## Summary
- Split the double meaning of `sessions.starts_at` (reserved start time at enqueue vs. actual execution start at the RUNNING transition) into a new `requested_starts_at` column, so the reservation is no longer lost when a batch session starts running.
- The reserved-batch validator now reads `requested_starts_at` and compares it against a DB-sourced snapshot timestamp (`SystemSnapshot.observed_at`) instead of per-server clocks; relative `starts_at` offsets are also resolved against DB time at enqueue (HA clock-skew safety).
- Alembic migration adds the column and backfills batch sessions that have not reached RUNNING yet (already-running/terminated sessions have lost the reservation, left NULL).

## Test plan
- [x] Reserved-batch validator hold/release scenarios with a fixed snapshot time, including the exact-boundary case
- [x] RUNNING transition writes `starts_at` without touching `requested_starts_at` (real-DB test)
- [x] Migration upgrade/downgrade verified against a scratch Postgres DB (9 cases across statuses/types)
- [x] Existing idle session-lifetime / fair-share tests unchanged and green

Resolves BA-6994

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by jopemachine** [comment]
News fragment is missing

File: src/ai/backend/manager/repositories/scheduler/creators.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def build_row(self) -> SessionRow:
             )
 
         main_mounts = list(main_kernel.vfolder_mounts) if main_kernel else []
-        session_starts_at = main_kernel.execution_spec.starts_at if main_kernel else None
+        session_requested_starts_at = main_kernel.execution_spec.starts_at if main_kernel else None
```
Comment (seedspirit) [inline src/ai/backend/manager/repositories/scheduler/creators.py:172]: Should this part also be changed to "session request starts at"? I was just wondering if "starts at" might be more accurate.

Comment (HyeockJinKim) [reply src/ai/backend/manager/repositories/scheduler/creators.py:172]: The "starts at" value combines two different concepts—the time the current session is running and the time the batch session is scheduled to run—which causes confusion between the concepts of scheduling and execution time.
Therefore, for this part, we treat the value specified in `enqueue` as pertaining to the batch session and proceed accordingly.

**Review by seedspirit — APPROVED** [review]
